src/resources/ECForms.ts

```diff
@@ -0,0 +1,121 @@
+/**
+ *  @module Resources/ECForms
+ */
+import { AuthParams, ResponseCallback, SendData } from "../api/RestAPI";
+
+import { AmountWithCurrency } from "./common/types";
+import { CRUDResource } from "./CRUDResource";
+import { InstallmentPlan, SubscriptionPeriod } from "./Subscriptions";
+import { TransactionTokenType, UsageLimit } from "./TransactionTokens";
+
+type BaseMetadata = Record<string, string>;
+
+export enum CheckoutType {
+    PAYMENT = "payment",
+    TOKEN = "token",
+}
+
+export enum ECFormCustomFieldType {
+    STRING = "string",
+    SELECT = "select",
+}
+
+export type ECFormCustomField = {
+    key: string;
+    label: string;
+    type: ECFormCustomFieldType;
+    required?: boolean;
+    options?: string[];
+};
+
+export type ECFormItem<Metadata = BaseMetadata> = {
+    /* EC Form data */
+    id: string;
+    storeId: string;
+    name: string;
+    appTokenId: string;
+    createdOn: string;
+
+    /* Redirects */
+    successRedirectUrl?: string;
+    failureRedirectUrl?: string;
+
+    /* Charge and token data */
+    appId: string;
+    checkout: CheckoutType;
+    paymentType: PaymentShippingType;
+    tokenType: TransactionTokenType;
+    univapayCustomerId: string | null;
+    capture: boolean;
+    captureAt?: string | null;
```

Comment:
this `univapayCustomerId: string | null`, it is `null`-able. Does that mean this is optional?

Reply:
`univapayCustomerId: string | null,` means that we expect `univapayCustomerId` to always be a property in the object that can be a string or `null` when it does not exist 🙂 

I may review the parameters in another PR to remove the one we wont use in a first time. Just wanted to keep it simple for this PR as this was moved from the private SDK.